clients/privacy-center/packages/fides-consent/src/lib/consent-banner.ts

```diff
@@ -0,0 +1,553 @@
+import {
+  CookieKeyConsent,
+  hasSavedConsentCookie,
+  setConsentCookieAcceptAll,
+  setConsentCookieRejectAll,
+} from "./cookie";
+
+export type ConsentBannerOptions = {
+  // Whether or not debug log statements should be enabled
+  debug?: boolean
+
+  // API URL to use for obtaining user geolocation. Must be provided if isGeolocationEnabled = true
+  // Expects this API to accept a basic HTTP GET and return a UserGeolocation response, e.g.
+  // ```bash
+  // curl "https://example-geolocation.com/getlocation?api_key=123"
+  // {"country":"US","ip":"192.168.0.1:1234","location":"US-NY","region":"NY"}
+  // ```
+  geolocationApiUrl?: string
+
+  // Whether or not the banner should be globally disabled
+  isDisabled?: boolean 
+
+  // Whether user geolocation should be enabled. Requires geolocationApiUrl
+  isGeolocationEnabled?: boolean
+
+  // List of country codes where the banner should be enabled. Requires isGeolocationEnabled = true
+  isEnabledCountries?: string[]
```

Comment:
Thinking ahead a bit: I think all the "geolocation" options should be reorganized into a sub-object like:
```
geolocation?: {
  isEnabled?: boolean
  isEnabledCountries?: string[]
  apiUrl?: string
}
```